### DISCUSSION TITLE
Firebase

### DISCUSSION THREAD

**Discussion by rechandler** [nextauthjs/next-auth#723] [Help]
**Your question**
Are there any examples or is it possible to set up a custom provider with Firebase Auth?

**What are you trying to do**
This library seems awesome and I'd love to user firebase auth but I'm not sure its possible. I can get Next to work with Firebase, but I'd prefer to use this library. Currently its set up to validate the firebase token on the server, but this means my entire app is now server side and not a mixture of static and ssr.

**Comment by revskill10** [comment]
I need firebase authentication to leverage firebase storage, too.

**Comment by CodenameCookie** [comment]
Did you ever find a good example of how to make NextAuth work with Firebase. I may be able to write the Firebase functions to work but its not going to be easy.

**Comment by rechandler** [comment]
You can follow the firebase docs and get it working, however it's going to prevent pre rendering of every page  that needs authentication, which in my app was every page.

**Comment by RonHouben** [comment]
I implemented a custom firebase adapter for a hobby project.
It's written in TS though but feel free to use it:
https://github.com/RonHouben/nextjs-tailwindcss-todo-app/blob/c55af99329f5206d31eec8f87405873e111895c6/lib/firebaseAdapter.ts

I don't have the time to translate it back to JS and create a PR to put it in the examples of the next-auth repo.
If someone would be so kind to do that we all help each other out a bit :)

**Reply by balazsorban44** [reply]
that's awesome! I think our new adapters repo would be a perfect place for this: https://github.com/nextauthjs/adapters

if you don't have the time and really don't mind, I might create a "good first issue" on adding it to the repo!

It could then just stay in TypeScript, since we would release it as a separat module: `@next-auth/firebase-adapter`

**Reply by RonHouben** [reply]
> that's awesome! I think our new adapters repo would be a perfect place for this: https://github.com/nextauthjs/adapters
> 
> if you don't have the time and really don't mind, I might create a "good first issue" on adding it to the repo!
> 
> It could then just stay in TypeScript, since we would release it as a separat module: `@next-auth/firebase-adapter`

Of course feel free to create an issue for it and let it get published however you feel fit :).
#SharingIsCaring

**Reply by balazsorban44** [reply]
Dankjewel @RonHouben! I opened the following "good first issue": https://github.com/nextauthjs/next-auth/issues/1222, and was just wondering. If someone from the community decides to try to add it, would you be open to helping them out by answering questions? Please don't feel any pressure, I totally understand if not. I was just thinking since it is your code, you may have some opinions which you would like to be present.
